### DISCUSSION TITLE
How do I install an extension found in a GitHub subdirectory?

### DISCUSSION THREAD

**Discussion by coatless** [quarto-dev/quarto-cli#8353] [Q&A]
### Description

I'm trying to have a repository setup that looks like:

```sh
.
|- lib/
|- quarto/
|  |- _extension/mynestedextension
|- r/
|- python/
|_ README.md
```

Is there a way I can direct the `quarto install` or `quarto add` command to retrieve the extension in `quarto/` folder?

It looks like there was a prior discussion in #1440 with no official resolution. Was subdirectory installation added or is it on the backburner?

/cc @gadenbuie

**Comment by cderv** [comment]
As mentioned in https://quarto.org/docs/extensions/distributing.html#github-distribution you should be able to install from subdirectory

> You can also add an extension from a subdirectory of a GitHub repository. For example, here we install two different extensions from the cooltools/icons repository:

````bash
quarto add cooltools/icons/fontawesome
quarto add cooltools/icons/iconify
````

We document it so I believe it has been added; Did you try subdirectory install already ?

**Reply by mcanouil** [reply]
Last time I checked (when I asked for this), it was working.
- https://github.com/quarto-dev/quarto-cli/issues/1836

**Reply by gadenbuie** [reply]
Does the subdirectory need to include an `_extensions` folder? If I try with a non-existent subdirectory, the command fails but the error message seems to indicate that `quarto add` automatically adds the `_extensions` directory to the path:

```
❯ quarto add gadenbuie/countdown/quarto/
[...snip...]
ERROR: NotFound: No such file or directory (os error 2), readdir '/var/folders/yl/0xhq7c7n3z7_9hp1356kctbr0000gp/T/quarto-sessiona0c35d21/fbf13978/a8024b69/archive/countdown-main/quarto/_extensions'

NotFound: No such file or directory (os error 2), readdir '/var/folders/yl/0xhq7c7n3z7_9hp1356kctbr0000gp/T/quarto-sessiona0c35d21/fbf13978/a8024b69/archive/countdown-main/quarto/_extensions'
```

It'd be nice to be able to install a specific extensions from within an `_extensions` folder, for example I have [a panelset extension in gadenbuie/xaringanExtra/_extensions/panelset](https://github.com/gadenbuie/xaringanExtra/tree/main/_extensions/panelset) but this doesn't work:

```
❯ quarto add gadenbuie/xaringanExtra/_extensions/panelset
[...snip...]
ERROR: NotFound: No such file or directory (os error 2), readdir '/var/folders/yl/0xhq7c7n3z7_9hp1356kctbr0000gp/T/quarto-sessiona72adb03/d987109b/b44a462a/archive/xaringanExtra-main/_extensions/panelset/_extensions'

NotFound: No such file or directory (os error 2), readdir '/var/folders/yl/0xhq7c7n3z7_9hp1356kctbr0000gp/T/quarto-sessiona72adb03/d987109b/b44a462a/archive/xaringanExtra-main/_extensions/panelset/_extensions'
```

**Reply by mcanouil** [reply]
cc @dragonstyle

**Reply by dragonstyle** [reply]
> Does the subdirectory need to include an _extensions folder?

Yeah, it does. We basically install from the repo by getting a copy of the repo (and subdirectory or branch) and then installing files that are present in the `_extensions` subdirectory. 

> It'd be nice to be able to install a specific extensions from within an _extensions folder.

There isn't a way to reach into an `_extensions` subdirectory and install just one of those elements - all of the extensions within the folder are considered a part of the extension which is being installed. In that case, the only extension present is the panelset, so if you use `quarto add gadenbuie/xaringanExtra` it will just install the panelset.

**Reply by gadenbuie** [reply]
Thanks @dragonstyle! And yeah, xaringanExtra only has a single extension for now but who knows, that could change in the future 😄
